diffulex/strategy/fast_dllm_v2/engine/sequence.py

```
@@ -0,0 +1,279 @@
+        offset = self.global_start_id - self.size * sum(block.is_to_cache for block in self.seq.diffusion_blocks)
+        return [mask_id + offset for mask_id in self.local_mask_token_ids]
+    
+
+@AutoSequence.register("fast_dllm_v2", is_default=True)
+class FDV2Sequence(SequenceBase):
+    """Sequence implementation tailored for diffusion-based decoding."""
+
+    def __init__(
+        self,
+        token_ids: list[int],
+        sampling_params: SamplingParams = SamplingParams(),
```

Ruff (0.14.14)

[warning] Do not perform function call `SamplingParams` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)